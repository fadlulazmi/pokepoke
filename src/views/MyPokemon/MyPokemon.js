import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {removeFavorite, getFavorite} from '../../store/action'
import Card from '../../components/Card'
import Button from '../../components/Button'
import './styles.scoped.css'

export default function MyPokemon() {
  const dispatch = useDispatch()
  const { myPokemon } = useSelector(state => state)
  useEffect(() => {
    dispatch(getFavorite())
  }, [dispatch])
  return (
    <div className="container">
      { 
        myPokemon.length > 0 
          ? myPokemon.map(el => (
            <div key={el.id}>
              <Card
                key={el.name}
                name={el.name}
                counter={el.owned}
              />
              <Button onClick={() => dispatch(removeFavorite(el.id))} text="REMOVE"/>
            </div>
          ))
          : <p>Let's catch POKEMON !</p>
      }
    </div>
  )
}
