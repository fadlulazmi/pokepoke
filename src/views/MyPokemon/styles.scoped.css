.favorite > p {
  color: pink;
}

.favorite > div {
  display: flex;
  flex-direction: column;
}

.favorite > div > .card {
  margin-bottom: 0;
  border-radius: 10px 10px 0 0;
}

.favorite > div > .button {
  margin: 0 10px;
  border-radius: 0 0 10px 10px;
  height: 30px;
  color: pink;
  font-weight: 600;
}