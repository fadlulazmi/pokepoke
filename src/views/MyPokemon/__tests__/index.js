import MyPokemon from '../MyPokemon'
import ShallowRenderer from 'react-test-renderer/shallow';
import { useSelector } from 'react-redux';


describe('src/views/MyPokemon', () => {
  test('should be rendered correctly', () => {
    useSelector.mockImplementationOnce(fn => {
      fn()
      return {
        myPokemon: []
      }
    })
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<MyPokemon />);
    expect(tree).toMatchSnapshot();
  })
})
