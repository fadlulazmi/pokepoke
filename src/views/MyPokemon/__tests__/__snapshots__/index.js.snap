// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/MyPokemon should be rendered correctly 1`] = `
<div
  className="favorite"
>
  <p>
    Let's catch POKEMON !
  </p>
</div>
`;
